src/core/server/rendering/rendering_service.test.ts

```diff
@@ -137,6 +137,29 @@ describe('RenderingService', () => {
     });
   });
 
+  describe('isResourceValid()', () => {
```

Comment:
Can you add functional tests to accompany the [existing custom branding functional tests](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/5edf03062d40a92385070f53904b2344c0ffd351/test/functional/apps/visualize/_custom_branding.js#L55)?